### PR TITLE
Uart fifo generic

### PR DESCRIPTION
**Pull request by koenenwmn**

### CODE REVIEW

**Comment by koenenwmn** [comment]
Final testing is still running, don't merge just yet.

**Comment by koenenwmn** [comment]
Should work now.

**Comment by imphil** [comment]
- I've pushed a couple more commits with minor cleanups and verified that the demo still works on Nexys 4 DDR. This needs to be squashed before merging, but leaving it as separate commits for now for easier collaboration.
- However, it does not work on the VC108 board as I got it from your desk. 
  - Are there any special settings necessary? (Vivado, switches on the board, etc.)
  - Which USB port should be the right one (higher or lower numbered?) Is there a reliable way to know which one of the two serial ports are the right ones?

File: src/backend_uart/logic/demo/nexys4ddr/README.md
Comment (wallento) [inline src/backend_uart/logic/demo/nexys4ddr/README.md:29]: UART is still set by default, right? Just clarification?

**Review by wallento — APPROVED** [review]

File: src/backend_uart/logic/demo/nexys4ddr/README.md
Comment (imphil) [reply src/backend_uart/logic/demo/nexys4ddr/README.md:29]: yes, it is. Nothing changed in this regard, just made it explicit to avoid breaking the docs if we should ever change the defaults.

**Pull request merged**
